docs/next/util/useVersion.ts

```diff
@@ -1,82 +1,25 @@
 import MAP_VERSION_TO_LINK from '../.versioned_content/_versions_with_static_links.json';
 
 export const latestVersion = ALL_VERSIONS[ALL_VERSIONS.length - 1];
-export const defaultVersion = latestVersion; // always point the default version to master
 
 export function getOlderVersions() {
   // exclude latest version which will be the current site. sort by version desc
   return MAP_VERSION_TO_LINK.slice(0, -1).sort((a, b) => (a.version < b.version ? 1 : -1));
 }
 
 // only hide version notice in production
-export let showVersionNotice = true;
+let showVersionNotice = true;
 if (process.env.NEXT_PUBLIC_VERCEL_ENV === 'production') {
-  // We use NEXT_PUBLIC_VERCEL_ENV to tell whether it's in production or not
+  // We use NEXT_PUBLIC_VERCEL_ENV to tell whether it's in production or not, because:
   // * NEXT_PUBLIC_VERCEL_ENV is exposed to the browser
-  // * Vercel previews have NODE_ENV === "production"
-  showVersionNotice = false;
-} else if (process.env.NODE_ENV === 'production') {
-  // for testing
+  // * Vercel previews have NODE_ENV === "production", so we can't rely on that.
+  //     As going forward, we are relying on Vercel's previews to version the older docs, we need to
+  //     make sure that when a user lands on the older docs (aka previews), we have the version
+  //     notice as reminder that they are looking at older docs.
   showVersionNotice = false;
 }
 
-export function normalizeVersionPath(
-  asPath: string,
-  versions?: string[],
-): {
-  version?: string;
-  asPath: string;
-  asPathWithoutAnchor: string;
-  versions: string[];
-  defaultVersion: string;
-  latestVersion: string;
-} {
-  let detectedVersion: string = defaultVersion;
-  // first item will be empty string from splitting at first char
-  const pathnameParts = asPath.split('/');
-
-  (versions || []).some((version) => {
-    if (pathnameParts[1].toLowerCase() === version.toLowerCase()) {
-      detectedVersion = version;
-      pathnameParts.splice(1, 1);
-      asPath = pathnameParts.join('/') || '/';
-      return true;
-    }
-    return false;
-  });
-
-  let asPathWithoutAnchor = asPath;
-  if (asPathWithoutAnchor.indexOf('#') > 0) {
-    asPathWithoutAnchor = asPath.substring(0, asPath.indexOf('#'));
-  }
-
-  // sort release versions by latest - we assume `ALL_VERSIONS` starts with master, and then
-  // the following versions are sorted from oldest to latest.
-  const sortedVersions = ALL_VERSIONS.slice(0, 1).concat(ALL_VERSIONS.slice(1).reverse());
-
-  return {
-    asPath,
-    asPathWithoutAnchor,
-    version: detectedVersion,
-    versions: sortedVersions,
-    defaultVersion,
-    latestVersion,
-  };
-}
-
 export const useVersion = () => {
-  const router = useRouter();
-
-  const [asPath, setAsPath] = useState('/');
-
-  useEffect(() => {
-    if (router.isReady) {
-      setAsPath(router.asPath);
-    }
-  }, [router]);
-  return normalizeVersionPath(asPath, ALL_VERSIONS);
+  return {version: latestVersion, showVersionNotice};
```

Comment:
Because this is no longer a client-state-dependent react hook (via `useRouter`), IMO `useVersion` (`use*` being a React hook convention) should go away entirely and there should just be exported constants available at build time:

- `LATEST_VERSION`
- `SHOW_VERSION_NOTICE`

Reply:
good call. will update.

Reply:
also updated `useNavigation` cos that's no longer using hooks either.